### PR TITLE
Add ethereum delegated manager deploy script [SIM-64]

### PR DESCRIPTION
**Pull request by pblivin0x**
Notes
- Add deploy script for `ManagerCore`, `DelegatedManagerFactory`, `IssuanceExtension`, `StreamingFeeSplitExtension`, and `TradeExtension`
- Add set-protocol-v2 dependency addresses for `SetTokenCreator`, `IssuanceModule`, `StreamingFeeModule`, and `TradeModule`

### CODE REVIEW

File: ethereum/deployments/utils/dependencies.ts [hunk 1]
```diff
@@ -411,6 +411,34 @@ export default {
...
+    },
+    42: "0xE038E59DEEC8657d105B6a3Fb5040b3a6189Dd51",
+  },
+  TRADE_MODULE: {
+    1: {
+      staging: "0x45D67b9dbEA9bd51ED2B67832addEAF839628fAa",
+      production: "0x90F765F63E7DC5aE97d6c576BF693FB6AF41C129"
+    },
+    42: "0xC93c8CDE0eDf4963ea1eea156099B285A945210a",
+  },
```
Comment (cgewecke) [inline ethereum/deployments/utils/dependencies.ts:441]: Double-checked these against outputs in set-v2-deployments, all LGTM.

File: ethereum/deploy/002_delegated_manager_system.ts [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+  const streamingFeeModuleAddress = await findDependency(STREAMING_FEE_MODULE);
+  await deployStreamingFeeSplitExtension();
+  const streamingFeeSplitExtensionAddress = await getContractAddress(CONTRACT_NAMES.STREAMING_FEE_SPLIT_EXTENSION);
+
+  const tradeModuleAddress = await findDependency(TRADE_MODULE);
+  await deployTradeExtension();
+  const tradeExtensionAddress = await getContractAddress(CONTRACT_NAMES.TRADE_EXTENSION);
+
+  await initializeManagerCore();
```
Comment (cgewecke) [inline ethereum/deploy/002_delegated_manager_system.ts:69]: Do we need to add a [transferOwnershipToMultisig][1] statement right before this for the production setup, similar to what is done for modules at set-v2-deployments? ManagerCore is governance controlled. 

(Need to double-check with Brian about this)

[1]: https://github.com/SetProtocol/set-v2-deployments/blob/9819a65819266ee3b2e8987614b2c54d5d8624f3/optimism/deploy/006_perpV2_module.ts#L397-L429

**Review by cgewecke — COMMENTED** [review]
This all LGTM. Left an open question about whether we should transfer ManagerCore to a multisig for production. 

One other thing...the production outputs file for the Ethereum network is wrong and will crash (mea culpa) Could we sneak a fix for that in here? Instead of being `{}` it should be:

```js
{
  "state": {
    "network_key": "1-production",
    "human_friendly_name": "main-net-production",
    "network_id": 1,
    "last_deployment_stage": 0
  } ,
  "addresses": {},
  "transactions": {}
}
```

File: ethereum/deploy/002_delegated_manager_system.ts (hunk 2 above)
Comment (pblivin0x) [reply ethereum/deploy/002_delegated_manager_system.ts:69]: Added `ManagerCore` ownership transfer to multisig. Also removed unnecessary deferred transaction in `ManagerCore.initialize`.

**Review by cgewecke — APPROVED** [review]
💯

**Pull request merged**